torch/_functorch/vmap.py

```diff
@@ -149,14 +149,78 @@ def _process_batched_inputs(
 
 # Creates BatchedTensors for every Tensor in arg that should be batched.
 # Returns the (potentially) batched arguments and the batch_size.
+def _add_batch_dim_python_wrapper(self, batch_dim, level):
+    from torch._export.utils import _maybe_find_pre_dispatch_tf_mode_for_export
+    from torch.fx.experimental.proxy_tensor import (
+        _fetch_proxies_and_all_constant_flag,
+        PreDispatchTorchFunctionMode,
+        track_tensor_tree,
+    )
+
+    mode = _maybe_find_pre_dispatch_tf_mode_for_export()
+    res = _add_batch_dim(self, batch_dim, level)
+
+    if mode is None:
+        return res
+
+    assert isinstance(mode, PreDispatchTorchFunctionMode)
+
+    tracer = mode.tracer
+
+    _, proxies, _ = _fetch_proxies_and_all_constant_flag(
```

Comment:
So.... please explain the BC problem here?

Reply:
Can you elaborate more on what broke when you just used `handle_torch_function`? In general we expect our torch function dispatch modes to be robust to new operators showing up

Reply:
+1, let's use handle_torch_function mode than run our own logic here

Reply:
Hmm here we are making sure to dispatch to pre-dispatch export torch function mode only.